exporter/collector/logs.go

```diff
@@ -43,20 +74,107 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld pdata.Logs) error {
+	mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
+	logPushErrors := []error{}
 	for i := 0; i < ld.ResourceLogs().Len(); i++ {
 		rl := ld.ResourceLogs().At(i)
-		mapper := &metricMapper{} // Refactor metricMapper to map MRs for logging?
 		mr, _ := mapper.resourceToMonitoredResource(rl.Resource())
 		for j := 0; j < rl.InstrumentationLibraryLogs().Len(); j++ {
 			ill := rl.InstrumentationLibraryLogs().At(j)
 			for k := 0; k < ill.LogRecords().Len(); k++ {
 				log := ill.LogRecords().At(k)
-				l.logger.Log(logging.Entry{
-					Resource: mr,
-					Payload:  log.Body().AsString(),
-				})
+
+				entry, err := l.mapper.logToEntry(log, mr)
+				if err != nil {
+					logPushErrors = append(logPushErrors, err)
+				} else {
+					l.logger.Log(entry)
+				}
 			}
 		}
 	}
+	if len(logPushErrors) > 0 {
+		return multierr.Combine(logPushErrors...)
+	}
 	return nil
 }
+
+func (l logMapper) logToEntry(
+	log pdata.LogRecord,
+	mr *monitoredres.MonitoredResource,
+) (logging.Entry, error) {
+	entry := logging.Entry{
+		Resource: mr,
+	}
+
+	logBody := log.Body().BytesVal()
+	if len(logBody) > 0 {
+		entry.Payload = json.RawMessage(logBody)
+	}
+
+	httpRequestAttr, ok := log.Attributes().Get(HTTPRequestAttributeKey)
+	if ok {
+		httpRequest, err := l.parseHTTPRequest(httpRequestAttr.BytesVal())
+		if err != nil {
+			l.obs.log.Debug("Unable to parse httpRequest", zap.Error(err))
+		}
+		entry.HTTPRequest = httpRequest
+	}
+
+	return entry, nil
+}
+
+// JSON keys derived from:
+// https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#httprequest
+type httpRequestLog struct {
+	RequestMethod                  string `json:"requestMethod"`
+	RequestURL                     string `json:"requestUrl"`
+	RequestSize                    int64  `json:"requestSize,string"`
+	Status                         int    `json:"status,string"`
+	ResponseSize                   int64  `json:"responseSize,string"`
+	UserAgent                      string `json:"userAgent"`
+	RemoteIP                       string `json:"remoteIp"`
+	ServerIP                       string `json:"serverIp"`
+	Referer                        string `json:"referer"`
+	Latency                        string `json:"latency"`
+	CacheLookup                    bool   `json:"cacheLookup"`
+	CacheHit                       bool   `json:"cacheHit"`
+	CacheValidatedWithOriginServer bool   `json:"cacheValidatedWithOriginServer"`
+	CacheFillBytes                 int64  `json:"cacheFillBytes,string"`
+	Protocol                       string `json:"protocol"`
+}
+
+func (l logMapper) parseHTTPRequest(httpRequestAttr []byte) (*logging.HTTPRequest, error) {
```

Comment:
What if you pass in `httpRequestAttr.MapVal()` (instead of `BytesVal`), so you can access all the keys directly using `Get()` instead of needing to marshal to a new `httpRequestLog` struct?

Reply:
I'm trying to do this and it should work, but it leads to a ton of extra stuff casting and checking types. Might be better to use `mapstructure`, i.e.
- Get the `com.google.httpRequest` attribute 
- convert the map value to `map[string]interface{}`
- Unmarshal to `httpRequestLog` with `mapstructure`
That way we still have the nicety of a struct enforcing the types of the fields while not needing to do a json Unmarshal from bytes.

Reply:
Ah sorry, I didn't realize the level of encapsulation `MapVal()` had with its return value. I was hoping it would just be a more accessible map. We can mark this as a TODO for optimization then, the alternative doesn't seem to give much of a difference

Reply:
Added a TODO comment in [c0d53f7](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/344/commits/c0d53f791521bae806c4f308ab7c2cb4263fdc48)